src/nl/surf/eduhub_rio_mapper/job.clj

```diff
@@ -41,19 +43,25 @@
                                           ::rio/opleidingscode ::ooapi/type ::ooapi/id])]
     (logging/with-mdc log-context
       (log/infof "Started job %s, action %s, type %s, id %s" token action type id)
-      (try
-        (with-context trace-context
-          (case action
-            "delete" (delete! job)
-            "upsert" (update! job)))
-        (catch Exception ex
-          (let [error-id                   (UUID/randomUUID)
-                {:keys [phase retryable?]} (ex-data ex)]
-            (logging/log-exception ex error-id)
-            {:errors {:error-id      error-id
-                      :trace-context trace-context
-                      :phase         (or phase :unknown)
-                      :message       (ex-message ex)
-                      ;; we default to retrying, since that captures
-                      ;; all kinds of unexpected issues.
-                      :retryable?    (not= retryable? false)}}))))))
+      (binding [*http-messages* (if http-logging-enabled (atom []) nil)]
+        (try
+          (with-context trace-context
+            (let [result (case action
+                           "delete" (delete! job)
+                           "upsert" (update! job))]
+              (cond-> result
+                      *http-messages*
+                      (assoc :http-messages @*http-messages*))))
+          (catch Exception ex
+            (let [error-id                   (UUID/randomUUID)
+                  {:keys [phase retryable?]} (ex-data ex)]
+              (logging/log-exception ex error-id)
+              (cond-> {:errors        {:error-id      error-id
+                                       :trace-context trace-context
+                                       :phase         (or phase :unknown)
+                                       :message       (ex-message ex)
+                                       ;; we default to retrying, since that captures
+                                       ;; all kinds of unexpected issues.
+                                      :retryable?    (not= retryable? false)}}
+                      *http-messages*
```

Comment:
This is always true, because you do not deref the var

Reply:
@jelmerderonde  `*http-messages*` is `nil` when the logging is disabled. If logging is enabled but no requests were sent, you'd get an empty vector. I'm not sure if the current behaviour is wanted,  though...

Reply:
Ja je hebt gelijk. Alsnog lijkt het niet helemaal goed te gaan helaas.